src/backend/distributed/planner/merge_planner.c

```diff
@@ -1288,7 +1341,7 @@ SourceResultPartitionColumnIndex(Query *mergeQuery, List *sourceTargetList,
  * table or source query in USING clause.
  */
 RangeTblEntry *
-ExtractMergeSourceRangeTableEntry(Query *query)
+ExtractMergeSourceRangeTableEntry(Query *query, bool missingOk)
```

Comment:
I believe you are introducing the flag to tolerate a complex source other than simple table or a select query? For new comers it might be misleading as "missing source" is ok in some cases, which is not the intent here. May be name it something on the lines,` "complexSourceOk"` or `"joinSourceOk" `etc

Reply:
`joinSourceOk` looks reasonable, thanks